### ISSUE TITLE
Fatal merge

### ISSUE THREAD

**Issue by amis92** [BSData/wh40k#5204]
OP by @Mad-Spy:

Mad_Spy Yesterday at 9:51 AM
Something really weird has been going on with the repo. I can see a commit I made https://github.com/BSData/wh40k/commit/74b52ef6bb84c9ad380d60a160cbaddad2a9bd3d, which is supposed to have been released in 1.3.06, but there's no record of it on the actual file that was changed https://github.com/BSData/wh40k/commits/master/Chaos%20-%20FW%20Renegade%20and%20Heretics.cat? I had something similar with Daemons the other day as well.

---

Amadeusz Today at 3:28 PM
So, that's strange indeed.
I've launched an investigation into WTF is going on
It seems @Thairne 's Merge commit is the tipping point: https://github.com/BSData/wh40k/commit/c99e03fc786397fdac80aa32678a6e9b9d0488f6

When browsing files at this commit, the FW R&H file has rev38, while one of this merge commit's parents has the revision at 39: https://github.com/BSData/wh40k/commit/bd1ddff6be6ae564ab223f56051b7697d8a4f4e6

---

Amadeusz Today at 3:42 PM
So that's what was lost, maybe except DA: https://github.com/BSData/wh40k/compare/7ad2836..bd1ddff

---

Amadeusz Today at 3:44 PM
I think to be exact is that Thairne's merge commit took our master as the "other" parent and the commits are still in history but their changes were considered "not used"
I honestly have no idea why it shows up like that, it's got to be some weird kink of git, but that's how it currently looks like.
I've tried doing that merge again, so there was a conflict of revisions for DA file.

---

Amadeusz Today at 3:47 PM
@Thairne it'd seem you've just commited your file and discarded the changes of the other files during the merge xD Which resulted in them being "merged" in terms of linking commits but "discarded" in terms of the actual merge commit, resulting in effective revert

---

<details>
   <summary>Log of operations for re-merging "forgotten" commits</summary>

```
PS C:\Users\amadeusz.sadowski\BattleScribe\data\wh40k> git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
PS C:\Users\amadeusz.sadowski\BattleScribe\data\wh40k> git rebase -i fix/fatal-da-merge
Auto-merging Aeldari - Craftworlds.cat
CONFLICT (content): Merge conflict in Aeldari - Craftworlds.cat
error: could not apply 42bbaa7... Drain wording update

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".

Could not apply 42bbaa7... Drain wording update
PS C:\Users\amadeusz.sadowski\BattleScribe\data\wh40k> git rebase --continue
[detached HEAD 4213990] Drain wording update
 Author: Simon Porter <fastswimming@hotmail.com>
 1 file changed, 9 insertions(+), 9 deletions(-)
Auto-merging Chaos - Daemons.cat
CONFLICT (content): Merge conflict in Chaos - Daemons.cat
error: could not apply ade175b... Fixes #5151

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".

Could not apply ade175b... Fixes #5151
PS C:\Users\amadeusz.sadowski\BattleScribe\data\wh40k> git rebase --continue
[detached HEAD 25dbe72] Fixes #5151
 Author: Iain Launchbury <iainlaunchbury@gmail.com>
 1 file changed, 48 insertions(+), 48 deletions(-)
Auto-merging Imperium - Astra Militarum.cat
CONFLICT (content): Merge conflict in Imperium - Astra Militarum.cat
error: could not apply 2f6e1ef... Fixes #5142

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".

Could not apply 2f6e1ef... Fixes #5142
PS C:\Users\amadeusz.sadowski\BattleScribe\data\wh40k> git rebase --continue
[detached HEAD bfacd61] Fixes #5142
 Author: Iain Launchbury <iainlaunchbury@gmail.com>
 1 file changed, 60 insertions(+), 40 deletions(-)
Auto-merging T'au Empire.cat
CONFLICT (content): Merge conflict in T'au Empire.cat
error: could not apply 9c9458f... T'au: v46, fix profile type names via resave

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".

Could not apply 9c9458f... T'au: v46, fix profile type names via resave
PS C:\Users\amadeusz.sadowski\BattleScribe\data\wh40k> git rebase --continue
Auto-merging T'au Empire.cat
CONFLICT (content): Merge conflict in T'au Empire.cat
error: could not apply 549ad76... T'au fix Longstrike after FAQ addendum

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".

Could not apply 549ad76... T'au fix Longstrike after FAQ addendum
PS C:\Users\amadeusz.sadowski\BattleScribe\data\wh40k> git rebase --continue
[detached HEAD e502e77] T'au fix Longstrike after FAQ addendum
 1 file changed, 3 insertions(+), 2 deletions(-)
Auto-merging Chaos - Chaos Space Marines.cat
CONFLICT (content): Merge conflict in Chaos - Chaos Space Marines.cat
error: could not apply 863fc14... Fixes #5203

Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".

Could not apply 863fc14... Fixes #5203
PS C:\Users\amadeusz.sadowski\BattleScribe\data\wh40k> git rebase --continue
[detached HEAD 1f542e6] Fixes #5203
 Author: Iain Launchbury <iain.launchbury@oxfordcc.co.uk>
 1 file changed, 19 insertions(+), 14 deletions(-)
Successfully rebased and updated refs/heads/master.
PS C:\Users\amadeusz.sadowski\BattleScribe\data\wh40k> git status
On branch master
Your branch and 'origin/master' have diverged,
and have 48 and 50 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

nothing to commit, working tree clean
PS C:\Users\amadeusz.sadowski\BattleScribe\data\wh40k>
```

</details>

**Comment by Thairne** [comment]
Affected cats by above log:

Aeldari - Craftworlds.cat -  Drain wording update
Chaos - Daemons.cat - #5151
Imperium - Astra Militarum.cat - #5142
T'au Empire.cat  - T'au: v46, fix profile type names via resave, T'au fix Longstrike after FAQ addendum
Chaos - Chaos Space Marines.cat - #5203

**Comment by amis92** [comment]
Umm the log is mainly the list of commits done *after* fatal merge, that contained changes conflicting with the changes made earlier but lost in the bad merge.

**Comment by Thairne** [comment]
Had a quick look what my merge fucked over in the cats. The bigger ones I will do tomorrow if it helps since it's past midnight here.

Craftworlds:
Storm Serpent (PTest) entry link and entry got removed
Number of casts for "psyker" went from 2 to 1

Drukhari:
Might have wrong Range on Spirit Vortex (8 instead of 18)

CSM:
Plague Marines lack a hide statement (Brazen Beasts)
as does Chaos Lord on Palanquin of Nurgle and Chaos Lord on Disc of Tzeentch
Chaos Lord on Steed of Slaanesh also lacks that constraint and lost its condition group in that case
Noise and Rubric Marines also.. all those Brazen Beasts constraints
Codex under artefacts of chaos lost a condition group as did "Renegade" and "Black Legion (Vigilus)"

FW Heretic Astartes :
seems to have lost the Multi Melta under Weapons of the Helbrute I think

Renegades and Heretics:
"Renegade Militia Squad" lost its publication
"Renegade Militia Champion" has a wrong profile (4+ vs 5+ and a few others)

Servants of the Abyss:
Black Legionaire lost its profile

Custodes:
Telemon Caestus - some modifier got changed from 2 selection to 1

Blood Angels:
Relics of Baal list its condition group

Deathwatch:
Deathwatch Relic  lost its condition group

Daemons.cat:

Astra Militarum:

FW Adeptus Astartes:

Space Wolves:

T'au Empire:

**Comment by amis92** [comment]
@BSData/data-authors-wh40k can we not re-apply fixes manually until #5204 is done? Adding more duplication won't help with fixing the situation.

**Comment by amis92** [comment]
The process of me restoring the changes lost in a fatal merge commit:

<details>
<summary>Command line with comments</summary>

```sh
git branch old-master bd1ddff6be6ae564ab223f56051b7697d8a4f4e6 # create branch to track last commit before fatal merge
git branch da-change 7867236d759efbc9d8536c5cec7a3eb822eb231c # create branch to track DA commit that was the other parent of the fatal merge
git checkout da-change
git merge old-master
# resolve conflict in Dark Angels file, commit merge
git checkout master
git rebase old-master # the fun starts
# resolving conflict for 42bbaa7917270746c559504a60a40ab721da907a "Drain wording update" - take current change (higher revision)
git add .
git rebase --continue
# resolving conflict for ade175b3c53092b42129cec9fdde7aa396bb4cd9 "Fixes #5151" - take current change (higher revision) and incoming change for Keeper of Secrets wound track (1-4, 5-8, 9-16+)
git add .
git rebase --continue
# resolving conflict for 2f6e1ef290e4396edea0df5247013de7478b7ea2 "Fixes #5142" - take current change (higher revision), current change for removal of 'Huge knife' and 'Thunderous headbutt' profiles from unit level
git add .
git rebase --continue
# resolving conflict for 9c9458f1c9031611ac3bbc7cd60385fae960964a "T'au: v46, fix profile type names via resave" - take current change (removal of Stim Injector profile)
# results in no changes
git rebase --skip
# resolving conflict for 549ad76abfde920d3daf1388917cc6ebbd128623 "T'au fix Longstrike after FAQ addendum"- take incoming change for Fire Caste Exemplar profile (to Codex publication ref)
git add .
git rebase --continue
# resolving conflict for 863fc149211256cb43dba06bf9fed873922f5cea "Fixes #5203" - take current change *higher revision)
git add .
git rebase --continue
# resolving conflict for f12c0a34b5bea62f392f3cfb2d11e9c81acc5b17 "Fixes #5206" - take all incoming changes regarding conditions
git add .
git rebase --continue
# rebase done, our master now diverged from origin/master
git merge origin/master
# resolve T'au empire conflict - take current change (no Shield Generator profile)
git add .
git commit
# now we're good - we could force push this master, but we'll make it another way:
# - a new branch to track this good state
git checkout -b fix/fatal-merge
# - checkout master and reset master to origin/master
git checkout master
git reset --hard origin/master
# from master aligned with origin, create new branch
git checkout -b tmp
# - apply changes from fix/fatal-merge
git checkout fix/fatal-merge -- .
git commit
# - make this the new state of branch fix/fatal-merge
git checkout -B fix/fatal-merge
# - push the branch and do a PR
git push --set-upstream origin fix/fatal-merge
```

</details>
